In [21]:
import pandas as pd
import pymysql
from sklearn.metrics import r2_score, mean_squared_error, mean_absolute_error, explained_variance_score
import numpy as np
import matplotlib.pyplot as plt
import os

DB_USER = os.getenv('MYSQL_USER')
DB_PASSWORD = os.getenv('MYSQL_PASSWORD')
DB_NAME = os.getenv('MYSQL_DATABASE')
DB_HOST = os.getenv('MYSQL_HOST_NOTEBOOK', 'localhost')
DB_PORT = int(os.getenv('MYSQL_PORT_NOTEBOOK', 3308))
DB_TABLE_NAME = 'predictions' 



connection = pymysql.connect(host=DB_HOST,
                                 port=DB_PORT,
                                 user=DB_USER,
                                 password=DB_PASSWORD,
                                 database=DB_NAME,
                                 cursorclass=pymysql.cursors.DictCursor)
    
with connection.cursor() as cursor:
    sql_query = f"SELECT real_score, predicted_score FROM {DB_TABLE_NAME};"
    cursor.execute(sql_query)
    predictions_data = cursor.fetchall()
connection.close()

df_predictions = pd.DataFrame(predictions_data)


y_true = df_predictions['real_score']
y_pred = df_predictions['predicted_score']

r2 = r2_score(y_true, y_pred)
mae = mean_absolute_error(y_true, y_pred)
mse = mean_squared_error(y_true, y_pred)
rmse = np.sqrt(mse)
explained_variance = explained_variance_score(y_true, y_pred)

print("\n--- Métricas de Evaluación del Modelo ---")
print(f"R²: {r2:.4f}")
print(f"MAE: {mae:.4f}")
print(f"MSE: {mse:.4f}")
print(f"RMSE: {rmse:.4f}")
print(f"Varianza Explicada: {explained_variance:.4f}")


--- Métricas de Evaluación del Modelo ---
R²: 0.7795
MAE: 0.4056
MSE: 0.2682
RMSE: 0.5179
Varianza Explicada: 0.7810


Al RECALCULAR el R2 de los datos que llegaron desde la base de datos vemos que se reduce aproximadamente un 8% en comparación cuando lo evaluamos desde el mismo notebook de entrenamiento, por lo que el modelo nos da en general 77.95% de R2